src/Command/SelfUpdate.php

```diff
@@ -71,21 +72,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 0;
         }
 
+        $currentBuild = realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0];
+
         try {
-            $this->updateBuild($latestTag);
-        } catch (\PharException $e) {
+            $handler->handle(function ($latestBuild) use ($currentBuild) {
```

Comment:
I wonder if the separation of concerns made here is relevant or if the `SelfUpdate` handler should both request the new phar and replace the old one. I don't see much value having this in the command which is only here to gather inputs from the outside env and user and display informations. This callback does none of these.